server/lib/util/filterEvaluator.js

```diff
@@ -0,0 +1,91 @@
+const _ = require('lodash')
+
+module.exports = function ExpressionParser () {
+  const basicOperators = {
+    'eq': _.eq,
+    'gt': _.gt,
+    'gte': _.gte,
+    'lt': _.lt,
+    'lte': _.lte,
+    '!eq': (arg, other) => { return !_.eq(arg, other) },
+    '!gt': (arg, other) => { return !_.gt(arg, other) },
+    '!gte': (arg, other) => { return !_.gte(arg, other) },
+    '!lt': (arg, other) => { return !_.lt(arg, other) },
+    '!lte': (arg, other) => { return !_.lte(arg, other) },
+    'match': (str, regex) => {
+      if (typeof str === 'string' && typeof regex === 'string') {
+        return new RegExp(regex).test(str)
+      }
+      return false
+    }
+  }
+
+  const loopOperators = {
+    and: _.every,
+    or: _.some
+  }
+
+  function evalWithContext (expression, context) {
+    function evalExpression (expression, index) {
+      if (typeof expression !== 'object') {
+        throw Error('expression is not an object', expression)
+      }
+
+      const keys = Object.keys(expression)
+      if (keys.length !== 1) {
+        throw Error('expression should only have one key', JSON.stringify(expression))
+      }
+
+      let operator = keys[0]
+      let args = expression[operator]
+
+      let argsType = typeof args
+
+      if (argsType === 'string' || argsType === 'number' || argsType === 'boolean' || argsType === 'undefined') {
+        return evalString(operator, context) === evalString(args, context)
+      }
+
+      if (basicOperators[operator]) {
```

Comment:
Is not possible the operator has an attribute to tell this type? Then we could have here something as : 

```javascript
switch(operator.type) {
    case "loop":
        return ....
    case "basic":
        return ....
    default:
        throw Error('unkown operator ' + operator + ' in expression ' + JSON.stringify(expression))
}

```